cpu/cc2538/radio/cc2538_rf.c

```diff
@@ -38,40 +38,50 @@
-    {&SYS_CTRL_SCGCRFC,      0x01                     },
-    {&SYS_CTRL_DCGCRFC,      0x01                     },
-    {&RFCORE_XREG_CCACTRL0,  0xf8                     },
-    {&RFCORE_XREG_TXFILTCFG, 0x09                     },
-    {&RFCORE_XREG_AGCCTRL1,  0x15                     },
-    {&ANA_REGS_IVCTRL,       0x0b                     },
-    {&RFCORE_XREG_MDMTEST1,  0x08                     },
-    {&RFCORE_XREG_FSCAL1,    0x01                     },
-    {&RFCORE_XREG_RXCTRL,    0x3f                     },
-    {&RFCORE_XREG_MDMCTRL1,  0x14                     },
-    {&RFCORE_XREG_ADCTEST0,  0x10                     },
-    {&RFCORE_XREG_ADCTEST1,  0x0e                     },
-    {&RFCORE_XREG_ADCTEST2,  0x03                     },
-    {&RFCORE_XREG_CSPT,      0xff                     },
-    {&RFCORE_XREG_MDMCTRL0,  0x85                     },
-    {&RFCORE_XREG_FSCTRL,    0x55                     },
-    {&RFCORE_XREG_FRMCTRL0,  AUTOCRC | AUTOACK        },
-    {&RFCORE_XREG_FRMCTRL1,  0x00                     },
-    {&RFCORE_XREG_SRCMATCH,  0x00                     },
-    {&RFCORE_XREG_FIFOPCTRL, CC2538_RF_MAX_DATA_LEN   },
-    {&RFCORE_XREG_RFIRQM0,   FIFOP | RXPKTDONE        },
-#if IS_USED(MODULE_IEEE802154_RADIO_HAL)
-    {&RFCORE_XREG_RFIRQM1,   TXDONE | CSP_STOP        },
-#endif
-    {&RFCORE_XREG_RFERRM,    STROBE_ERR | TXUNDERF | TXOVERF | RXUNDERF | RXOVERF | NLOCK},
-    {NULL, 0},
-};
+/**
+ * @brief MAC timer period
+ *
+ * The period is set to the CSMA-CA Backoff Period Unit (20 symbols, 320 us).
+ * The system clock runs at 32 MHz. Thus, the timeout period is
+ * 320us * 32MHz = ~10738 (0x29F2)
+ */
+#define CCTEST_OBSSELX_EN           (BIT(7))
+
+static void _cc2538_setup_mac_timer(void)
+{
+    RFCORE_SFR_MTMSEL &= ~CC2538_SFR_MTMSEL_MASK;
+    /* Select timer period */
+    RFCORE_SFR_MTMSEL |= CC2538_SFR_MTMSEL_TIMER_P;
+
+    /* Fix timer to Backoff period */
+    RFCORE_SFR_MTM0 |= TIMER_PERIOD_LSB;
+    RFCORE_SFR_MTM1 |= TIMER_PERIOD_MSB;
+
+    RFCORE_SFR_MTMSEL &= ~CC2538_SFR_MTMSEL_MASK;
+    RFCORE_SFR_MTCTRL |= CC2538_MCTRL_SYNC_MASK | CC2538_MCTRL_RUN_MASK;
+}
+
+static void _cc2538_observable_signals(void)
+{
+    if (IS_USED(MODULE_CC2538_RF_OBS_SIG)) {
+        if (CONFIG_CC2538_RF_OBS_0 != disabled) {
+            RFCORE_XREG_RFC_OBS_CTRL0 = CONFIG_CC2538_RF_OBS_0;
+            *(&CCTEST_OBSSEL0 + CONFIG_CC2538_RF_OBS_SIG_0_PCX) = \
+                CCTEST_OBSSELX_EN | rfc_obs_sig0;
+        }
+        if (CONFIG_CC2538_RF_OBS_1 != disabled) {
+            RFCORE_XREG_RFC_OBS_CTRL1 = CONFIG_CC2538_RF_OBS_1;
+            *(&CCTEST_OBSSEL0 + CONFIG_CC2538_RF_OBS_SIG_1_PCX) = \
+                CCTEST_OBSSELX_EN | rfc_obs_sig1;
+        }
+        if (CONFIG_CC2538_RF_OBS_2 != disabled) {
+            RFCORE_XREG_RFC_OBS_CTRL2 = CONFIG_CC2538_RF_OBS_2;
+            *(&CCTEST_OBSSEL0 + CONFIG_CC2538_RF_OBS_SIG_2_PCX) = \
+                CCTEST_OBSSELX_EN | rfc_obs_sig2;
+        }
```

Comment:
Can you add a comment here what those signals indicate?

Reply:
This logic just selects which observable signal goes to what pin, what each signal inficates depends on what is assigned to `CONFIG_CC2538_RF_OBS_%`. I'll add a comment.